torch/nn/utils/parametrize.py

```diff
@@ -282,6 +283,15 @@ def _inject_new_class(module: Module) -> None:
     """
     cls = module.__class__
 
+    def default_deepcopy(self, memo=None):
+        # Just emulate a standard deepcopy procedure when __deepcopy__ doesn't exist in the current class.
+        memo = {} if memo is None else memo
+        replica = self.__new__(self.__class__)
+        memo[id(self)] = replica
+        for key, value in self.__dict__.items():
```

Comment:
You can do `replica.__dict__ = deepcopy(self.__dict__)`
You need also to handle slots by doing something like:
```python
  slots_to_save = copyreg._slotnames(self.__class__)  # type: ignore[attr-defined]
  for slot in slots_to_save:
      if hasattr(self, slot):
          setattr(replica, slot, deepcopy(getattr(self, slot), memo))
```

Reply:
1. Sure, replaced to `replica.__dict__ = deepcopy(self.__dict__, memo)` :heavy_check_mark: 
2. Added code of slots copying from `Tensor.__deepcopy__`.

The only question I have here is about slots registering. I thought that our modules isn't assumed to use `__slots__` and just rely on regular `__dict__` attributes storing. To explain what I mean here's a toy example:

```python
class Model(nn.Module):
    
    __slots__ = ["weight"]
    
    def __init__(self):
        super().__init__()
        self.weight = nn.Parameter(torch.tensor([1., 1., 1., 1.]), requires_grad=True)

model = Model()
model_copy = deepcopy(model)

```
This code raises an error: when `deepcopy` tries to restore object via `__setstate__` it can't do it because it's assumed that all our attributes are stored inside the `__dict__`.

I will be very grateful if you clarify the situation for me.

Reply:
I don't think parameters or buffers can be slots indeed. But users might use other attributes as slots.
Note that we are considering having some of the builtin attributes as slots as well.

Reply:
Got it, thank you!